dask/_task_spec.py

```diff
@@ -693,6 +751,61 @@ def is_coro(self):
         return self._is_coro
 
 
+class NestedContainer(Task):
+    klass: type
+    __slots__ = tuple(__annotations__)
+
+    def __init__(
+        self,
+        /,
+        *args: Any,
+        _dependencies: set | frozenset | None = None,
+        **kwargs: Any,
+    ):
+        super().__init__(
+            None,
+            self.to_container,
+            *args,
+            _dependencies=_dependencies,
+            **kwargs,
+        )
+
+    @classmethod
+    def to_container(cls, *args, **kwargs):
+        return cls.klass(args)
+
+
+class List(NestedContainer):
+    klass = list
+
+
+class Tuple(NestedContainer):
+    klass = tuple
+
+
+class Set(NestedContainer):
+    klass = set
+
+
+class Dict(NestedContainer):
+    klass = dict
```

Comment:
I ended up adding those convenience classes that implement the required logic to generate the appropriate container. These things are required to pass tasks or references in containers or nested containers. This should be an intentional opt-in for developers. user provided arguments could always be wrapped by this stuff, for example.

Reply:
@hendrikmakait you may want to review this

Reply:
This works for me. I'm wondering if the code where these are imported would be clearer if these had something like a `Task` suffix, but that can easily be changed later on.

Reply:
Yes I wondered the same thing but figured this is the responsibility of the importing code. This especially collides with the typing classes (that are discouraged from being used). I think this is the responsibility of the importing module. As long as there are no conflicts I suggest to keep the imports here as is